addon/components/file-dropzone/component.js

```diff
@@ -118,16 +118,30 @@ export default Ember.Component.extend({
   didDrop({ originalEvent: evt }) {
     // Testing support for dragging and dropping images
     // from other browser windows
-    let url = evt.dataTransfer.getData('text/uri-list');
-    let html = evt.dataTransfer.getData('text/html');
-    if (html) {
+    let supportsUrls, supportsHtml, html, url;
+
+    try {
+      html = evt.dataTransfer.getData('text/html');
+      supportsHtml = true;
```

Comment:
Makes sense! I thought you were saying to cache the actual logic at the top of the file, which left me scratching my head. (I spend roughly 50% of my day scratching my head, so this isn't out of the norm.)

See new commit 48a826d5ae12509dc4dd3d6693d75646497e04bc, which I agree is great for performance.